$OpenBSD: patch-src_backend_port_dynloader_openbsd_c,v 1.1 2011/04/07 13:54:16 pea Exp $

Remove old restrictions that no longer exist on OpenBSD.
Without these, PostgreSQL can now run on mips platforms.

--- src/backend/port/dynloader/openbsd.c.orig	Thu Apr  7 03:30:11 2011
+++ src/backend/port/dynloader/openbsd.c	Thu Apr  7 03:32:13 2011
@@ -57,28 +57,18 @@ BSD44_derived_dlerror(void)
 void *
 BSD44_derived_dlopen(const char *file, int num)
 {
-#if defined(__mips__)
-	snprintf(error_message, sizeof(error_message),
-			 "dlopen (%s) not supported", file);
-	return NULL;
-#else
 	void	   *vp;
 
 	if ((vp = dlopen((char *) file, num)) == NULL)
 		snprintf(error_message, sizeof(error_message),
 				 "dlopen (%s) failed: %s", file, dlerror());
 	return vp;
-#endif
 }
 
 void *
 BSD44_derived_dlsym(void *handle, const char *name)
 {
-#if defined(__mips__)
-	snprintf(error_message, sizeof(error_message),
-			 "dlsym (%s) failed", name);
-	return NULL;
-#elif defined(__ELF__)
+#if defined(__ELF__)
 	return dlsym(handle, name);
 #else
 	void	   *vp;
